compute/metadata/metadata.go

```diff
@@ -296,6 +296,7 @@ func (c *Client) getETag(suffix string) (value, etag string, err error) {
 		// being stable anyway.
 		host = metadataIP
 	}
+	suffix = strings.TrimLeft(suffix, "/")
 	u := "http://" + host + "/computeMetadata/v1/" + suffix
```

Comment:
We should probably use `path.Join` here instead of manually handling slashes.

Reply:
Suffix's can contain query parameters that can contain URLs, using `path.Join` will break users that are not URL encoding their parameters today. That is, if we path.Join on the whole suffix. I suppose we could split on `?`.... 

For example
```go
// suffix := "instance/service-accounts/default/identity?audience=http://example.com"
fmt.Println(path.Join(host, "/computeMetadata/v1/", suffix))
// Output: "169.254.169.254/computeMetadata/v1/instance/service-accounts/default/identity?audience=http:/example.com"
```

Maybe this is okay though as people should be encoding their query parameters... Thoughts?